tools/integration_tests/improved_run_e2e_tests.sh

```diff
@@ -509,17 +514,50 @@ test_package() {
   # This ensures spaces and special characters within arguments are handled correctly.
   local go_test_cmd=$(printf "%q " "${go_test_cmd_parts[@]}")
   
-  # Run the package test command
+  # Run the package test command and capture log output with runtime stats.
   local start=$SECONDS exit_code=0 
-  if ! eval "$go_test_cmd"; then
+  local log_file=$(mktemp)
+  # Ensure the temporary log file is removed on function exit.
+  trap 'rm -f "$log_file"' RETURN
+
+  if ! eval "$go_test_cmd" > "$log_file" 2>&1; then
     exit_code=1
   fi
   local end=$SECONDS
   # Add the package stats to the file.
   echo "${package_name} ${bucket_type} ${exit_code} ${start} ${end}" >> "$PACKAGE_RUNTIME_STATS"
+  # Generate Kokoro artifacts(log) files.
+  generate_test_log_artifacts "$log_file" "$package_name" "$bucket_type"
   return "$exit_code"
 }
 
+generate_test_log_artifacts() {
+  # If KOKORO_ARTIFACTS_DIR is not set, skip artifact generation.
+  if ! ${KOKORO_DIR_AVAILABLE} ; then
+    return 0
+  fi
+
+  if [[ $# -ne 3 ]]; then
+    log_error_locked "generate_test_log_artifacts() called with incorrect number of arguments."
+    return 1
+  fi
+
+  local output_dir="${KOKORO_ARTIFACTS_DIR}/${bucket_type}/${package_name}"
```

Comment:
bucket_type and package_name is local variable for this function, how are we catching these through passed flags?